contractcourt/chain_arbitrator.go

```diff
@@ -232,12 +232,10 @@ func newActiveChannelArbitrator(channel *channeldb.OpenChannel,
 		ShortChanID: channel.ShortChanID(),
 		BlockEpochs: blockEpoch,
 		ForceCloseChan: func() (*lnwallet.LocalForceCloseSummary, error) {
-			// First, we mark the channel as borked, this ensure
-			// that no new state transitions can happen, and also
-			// that the link won't be loaded into the switch.
-			if err := channel.MarkBorked(); err != nil {
-				return nil, err
-			}
+
+			// TODO: Must BORK first to avoid any updates? Or get a
+			// lock ensuring no updates won't be made before
+			// marking commitment broadcasted.
```

Comment:
We did this so the link wouldn't be loaded in again while we were force closing. If we remove this, then as we're force closing, then it's possible that the peer re-connects (after the disable below), and initiates a state transition.

Reply:
This is a bit tricky, since the two obvious ways here both have a gap:
1. First mark the channel borked, get the force close tx and write it to the DB:
* If we go down after marking it borked then we don't know at startup that we want to force close it.
2. Get the force close tx, then store it atomically to the db. 
* We don't really have a guarantee that no state updates happen between getting the tx and storing it to the db.

The first way to solve this that comes to mind, is for `MarkBorked` to take an "intention" argument (or just add a new state `MarkLocalForceClose`) such that we know on restart that we are to force close the channel.

Reply:
Go with new state

Reply:
When adding this new state, I realized that it is actually already encoded within the `ChannelArbitrator`'s state machine. When we trigger a force close it will commit to `StateBroadcastCommit`, and in this state mark the channel borked, get the force close tx, and then finally mark the commitment broadcasted. 

This means that the scenario 1) I outlined above is handled by the `ChannelArbitrator` itself, since after a restart it will be in `StateBroadcastCommit` and republish the force close tx. So we should be okay.